From: Daniel Veillard <veillard@redhat.com>
Date: Wed, 18 Jul 2012 17:39:56 +0800
Subject: Fix a failure to report xmlreader parsing failures

Related to https://bugzilla.gnome.org/show_bug.cgi?id=654567
the problem is that the provided patch failed to raise an error
on xmlTextReaderRead() return when an actual parsing error occured
---
 xmlreader.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/xmlreader.c b/xmlreader.c
index 77ac71b..10417ee 100644
--- a/xmlreader.c
+++ b/xmlreader.c
@@ -907,8 +907,11 @@ xmlTextReaderPushData(xmlTextReaderPtr reader) {
 	}
     }
     reader->state = oldstate;
-    if (reader->ctxt->wellFormed == 0)
+    if (reader->ctxt->wellFormed == 0) {
 	reader->mode = XML_TEXTREADER_MODE_EOF;
+        return(-1);
+    }
+
     return(0);
 }
 
-- 
